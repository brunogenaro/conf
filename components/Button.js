import React, { PureComponent } from 'react';

import Globals from '../Globals';

export default class extends PureComponent {
  render() {
    const { label, pattern } = this.props;

    return (
      <div className='background'>
        <div className='cont'>
          <h2>{label}</h2>
        </div>

        <style jsx>{`
          .cont {
            text-align: center;
            text-transform: uppercase;
            border-radius: 3;
            padding: 8px 40px;
            color: ${Globals.colors.darken};
            font-weight: bold;
            cursor: pointer;
            border: 1px solid ${Globals.colors.primary};

            > h2 {
              margin: 0;
              letterSpacing: 2
            }

            background-color: ${Globals.colors.primary};
            transition: ${Globals.transitions.primary};

            :hover {
              background-color: 'transparent';
              color: ${Globals.colors.primary};
            }

            @media(max-width: 720px) {
              margin: '5px';
            }
          }

          .background: {
            padding: 10,
            background-repeat: no-repeat;
            background-position: center;
            background-size: cover;
            margin: 30px;
          }
        `}</style>
      </div>
    );
  }
}
