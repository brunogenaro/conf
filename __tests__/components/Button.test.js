import React from 'react';
import renderer from 'react-test-renderer'
import Button from '../../components/Button.js';

describe('With Snapshot Testing', () => {
  it('Button Snapshot Testing', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
});