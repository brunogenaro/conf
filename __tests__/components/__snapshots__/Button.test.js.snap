// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Button Snapshot Testing 1`] = `
<div
  className="background"
  data-jsx={2406450419}
>
  <div
    className="cont"
    data-jsx={2406450419}
  >
    <h2
      data-jsx={2406450419}
    />
  </div>
</div>
`;
